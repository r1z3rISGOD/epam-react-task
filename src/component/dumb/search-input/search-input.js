"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchInput = void 0;
var React = require("react");
require("./search-input.scss");
var SearchInput = function (_a) {
    var setRequest = _a.setRequest, request = _a.request, setting = _a.setting, resultsSort = _a.resultsSort, fetchingAllFilms = _a.fetchingAllFilms;
    return (React.createElement("form", { className: 'search__form', onSubmit: function (event) {
            event.preventDefault();
            if (request) {
                fetchingAllFilms(request, setting, resultsSort);
            }
        } },
        React.createElement("input", { onChange: function (event) { return setRequest(event.target.value); }, type: "text", className: "search__input" })));
};
exports.SearchInput = SearchInput;
//# sourceMappingURL=search-input.js.map