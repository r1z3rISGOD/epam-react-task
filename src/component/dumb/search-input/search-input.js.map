{"version":3,"file":"search-input.js","sourceRoot":"","sources":["search-input.tsx"],"names":[],"mappings":";;;AAAA,6BAA8B;AAC9B,+BAA4B;AAWrB,IAAM,WAAW,GAAgC,UAAC,EAA+D;QAA7D,UAAU,gBAAA,EAAE,OAAO,aAAA,EAAE,OAAO,aAAA,EAAE,WAAW,iBAAA,EAAE,gBAAgB,sBAAA;IACpH,OAAO,CACG,8BAAM,SAAS,EAAC,cAAc,EAAC,QAAQ,EAAE,UAAC,KAAkC;YAC1E,KAAK,CAAC,cAAc,EAAE,CAAA;YACtB,IAAI,OAAO,EAAE;gBACX,gBAAgB,CAAC,OAAO,EAAE,OAAO,EAAE,WAAW,CAAC,CAAA;aAChD;QACH,CAAC;QACD,+BAAO,QAAQ,EAAE,UAAC,KAA2C,IAAK,OAAA,UAAU,CAAC,KAAK,CAAC,MAAM,CAAC,KAAK,CAAC,EAA9B,CAA8B,EAAE,IAAI,EAAC,MAAM,EAAC,SAAS,EAAC,eAAe,GAAE,CAClI,CAChB,CAAA;AACH,CAAC,CAAA;AAXY,QAAA,WAAW,eAWvB"}