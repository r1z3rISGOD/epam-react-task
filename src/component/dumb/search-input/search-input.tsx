import * as React from 'react'
import './search-input.scss'
import { changeSearchRequestedData, changeTypeOfSearch, putFetchedArrayToStore } from '../../../store/actions'
import { useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'
import { fetchData } from '../../services/fetchService'
import { FormEvent } from 'react'

interface SearchInputProps {
    setRequest: any
    request: string
    setting: string
    resultsSort: string
}

export const SearchInput : React.FC<SearchInputProps> = ({ setRequest, request, setting, resultsSort }) => {
  const history = useHistory()
  const dispatch = useDispatch()

  return (
            <form onSubmit={async (event : FormEvent<HTMLFormElement>) => {
              event.preventDefault()
              if (request) {
                dispatch(changeSearchRequestedData(request))
                dispatch(changeTypeOfSearch(setting))
                const response = await fetchData(request, setting, resultsSort)
                dispatch(putFetchedArrayToStore(response.data))
                history.push(`/search/?request=${request}`)
              }
            }}>
            <input onChange={(event : React.ChangeEvent<HTMLInputElement>) => setRequest(event.target.value)} type="text" className="search__input"/>
            </form>
  )
}
