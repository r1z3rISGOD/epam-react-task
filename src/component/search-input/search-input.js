import React from "react";
import "./search-input.scss"

export default function SearchInput() {
        return(
            <input type="text" className="search__input"/>
        )
}
