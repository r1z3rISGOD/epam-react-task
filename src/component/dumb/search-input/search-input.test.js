import { SearchInput } from './search-input.tsx'
import React from 'react'

const mockFetchingAllFilms = jest.fn(() => 1)
const mockSetRequest = jest.fn(() => 1)
const setUp = (request) => shallow(<SearchInput setRequest={mockSetRequest} request={request} setting='title' resultsSort='genres' fetchingAllFilms={mockFetchingAllFilms}/>)

it('search input 1', () => {
  const comp = setUp('harry')
  expect(comp).toMatchSnapshot()
  const form = comp.find('.search__form')
  form.simulate('submit', {
    preventDefault: () => {
    }
  })
  expect(mockFetchingAllFilms.mock.calls.length).toBe(1)
})

it('search input 2', () => {
  const comp = setUp()
  expect(comp).toMatchSnapshot()
  const input = comp.find('.search__input')
  input.simulate('change', {
    target: () => {

    }
  })
  expect(mockSetRequest.mock.calls.length).toBe(1)
})
