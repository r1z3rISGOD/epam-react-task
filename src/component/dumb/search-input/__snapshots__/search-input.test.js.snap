// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search input 1 1`] = `
<form
  className="search__form"
  onSubmit={[Function]}
>
  <input
    className="search__input"
    onChange={[Function]}
    type="text"
  />
</form>
`;

exports[`search input 2 1`] = `
<form
  className="search__form"
  onSubmit={[Function]}
>
  <input
    className="search__input"
    onChange={[Function]}
    type="text"
  />
</form>
`;
